### ISSUE TITLE
When updating an object which has been inserted in a deepObserved one, nothing happens

### ISSUE THREAD

**Issue by akarelas** [anywhichway/proxy-observe#16]
Shouldn't this code produce some output after the `-------` line?

```javascript
require('proxy-observe');

var scope = {};
var person = { name: 'Alex' };

scope = Object.deepObserve(scope, function(changeset) {
    console.log(changeset);
});

scope.person = person;
console.log('------------------');
scope.person.name = 'Kontogouris';
```

**Comment by devthejo** [comment]
You have to make it Observable by wrapping it in a Proxy object this way:

```
var scope = {};
var person = { name: 'Alex' };

person = ObjectObservable.create(person);

scope = Object.deepObserve(scope, function(changeset) {
    console.log(changeset);
});

scope.person = person;
console.log('------------------');
scope.person.name = 'Kontogouris';
```

**Comment by akarelas** [comment]
I tried it, and it threw an error: "ObjectObservable" not defined. Also your code doesn't contain that string.

**Comment by akarelas** [comment]
Did you just mean that this is how Object.observe used to work?

**Comment by anywhichway** [comment]
deepObserve is currently designed to work on objects that are part of the original observed object:

```
require('proxy-observe');

var scope = {};
var person = { name: 'Alex' };
scope.person = person;
scope = Object.deepObserve(scope, function(changeset) {
    console.log(changeset);
});
```

The module codebase could perhaps be modified so that if new additions are also objects the deepObserving would be extended to the new additions. In the meantime, you might be able to accomplish what you need by changing the deepObserver callback to something similar to the below (TOTALLY UNTESTED):

```
scope = Object.deepObserve(scope, function cb(changeset) {
    console.log(changeset);
   changeset.forEach((change) => {
     if(change.newValue && typeof(change.newValue)==="object") {
          let o = Object.observe(change.newValue,cb);
          scope[change.name] = o;
    }    
  });
});
```

Totally slammed with ReasonDB module at the moment, so if someone wants to try and make the core module change and submit a pull request, than would be great.

@surikat thanks for jumping in initially.

**Comment by vitormsilva** [comment]
Thanks in advance for this magical implementation.

I have the same problem.

I try what you said @anywhichway , and it works, but I wanted more, so I've changed the `Object.observe` to` Object.deepObserve` to get all the `keypath` reference.. but instead of that... I only have the current key which was changed.

i have a [jsbin](https://jsbin.com/nucovi/edit?js,console#J:L273), i don't know if I'm doing something wrong..

Could someone help me .. or give me some hints?
